poc/vdaf_mastic.py

```diff
@@ -0,0 +1,370 @@
+                                                  nonce)
+
+        # Compute the FLP verifier share, if applicable.
+        verifier_share = None
+        if cls.do_range_check(agg_param):
+            # Evaluate the VIDPF at each child of the root node.
+            #
+            # It is sufficient to check the proof over the sum of the outputs,
+            # since the one-hot verifiability property of VIDPF ensures that
+            # exactly one of the children is equal to the encoded `beta` (and
+            # the other is equal to `0`).
+            #
+            # It is sufficient to check the proof just once, since the path
+            # verifiability property of VIDPF ensures that the same `beta` is
+            # propagated along the entire `alpha` path.
+            #
+            # Implementation note: This invocation of the VIDPF is redundant.
+            # We evaluate at least one (and likely both) of these prefixes
+            # during the main invocation below.
+            #
+            # TODO(cjpatton) Evaluate at the root instead, as long as this
+            # doesn't add communication cost.
+            (out_share, _level0_proof) = cls.Vidpf.eval(agg_id,
+                                                        correction_words,
+                                                        init_seed,
+                                                        0,
+                                                        [0, 1],
+                                                        cs_proofs,
+                                                        cls.ROOT_PROOF,
+                                                        nonce)
+            meas_share = vec_add(out_share[0], out_share[1])
+
+            query_rand = cls.Xof.expand_into_vec(
+                cls.Flp.Field,
+                verify_key,
+                cls.domain_separation_tag(USAGE_QUERY_RAND),
+                nonce, # TODO(cjpatton) Consider binding to the VIDPF mode
+                cls.Flp.QUERY_RAND_LEN,
+            )
+
+            verifier_share = cls.Flp.query(meas_share,
+                                           proof_share,
+                                           query_rand,
+                                           [], # joint_rand
+                                           cls.SHARES)
+
+        prep_state = []
+        for val_share in out_share:
+            prep_state += cls.Flp.truncate(val_share)
+        prep_share = (level_proof, verifier_share)
+        return (prep_state, prep_share)
+
+    @classmethod
+    def prep_shares_to_prep(cls, agg_param, prep_shares):
+        if len(prep_shares) != 2:
+            raise ValueError('unexpected number of prep shares')
+
+        # Verify the VIDPF output.
+        (level_proof_0, verifier_share_0) = prep_shares[0]
+        (level_proof_1, verifier_share_1) = prep_shares[1]
+        if level_proof_0 != level_proof_1:
+            raise Exception('output vector is not one-hot')
```

Comment:
How about?
```python
raise Exception('Path Verifiability Error: The output vector is not one-hot.')
```

Reply:
This isn't Path Verifiability though, right? This is One-hot Verifiability.

Reply:
Correct, I meant *One-hot* verifiability.